web/source/build.sh

```diff
@@ -297,10 +298,18 @@ while [[ $# -gt 0 ]] ; do
         -clean)
             clean
             ;;
+        -upload-sentry)
+            UPLOAD_SENTRY=true
+            ;;
     esac
     shift # past argument
 done
 
+# No Sentry uploading if not a release build and not directly specified.
+if [ $UPLOAD_SENTRY = false ]; then
+    DO_NATIVE_SENTRY=false
```

Comment:
I'm not entirely clear on the difference between $UPLOAD_SENTRY and $DO_NATIVE_SENTRY?

Reply:
`$DO_NATIVE_SENTRY` already existed.

Sentry management for KMW is a bit tricky, as there are two separate versions to upload - `native` and `embedded`.  Embedded uploads are super-simple to manage - if we want an upload and we've compiled embedded-mode, upload it.

'Native' mode is less straightforward, at least in a development environment.  What behavior would we expect from `./build.sh -ui -upload-sentry`?  This command doesn't build the KeymanEngine - just the UI modules.  Alternatively, `./build.sh -web -upload-sentry` doesn't build the UI modules, but just the 'native' web engine.  What role does `-upload-sentry` have then?

The solution I opted for - so long as at least one of `-web` and `-ui` are (effectively) enabled, `$DO_NATIVE_SENTRY` is set - we've just recompiled at least part of native Web and thus have new artifacts to send to the server, even if part of them may be old.

Reply:
Hmm. Seems to me that uploading to Sentry is really only needed for a full build. I can't see a normal scenario where we'd be wanting to upload from a partial build -- testing the Sentry integration during development work does not seem to me to be a good scenario for this pattern.

I'd prefer to simplify the logic!

Reply:
> I can't see a normal scenario where we'd be wanting to upload from a partial build -- testing the Sentry integration during development work does not seem to me to be a good scenario for this pattern.

At a minimum, I'm frequently chopping off the 'embedded' part of builds during development.  When minification is at play, cutting both the 'embedded' and 'ui' portions generally chops a minute or so off of build times.

Of course, the _fastest_ builds come with the `-test` parameter in place, but that doesn't support Sentry uploads because it doesn't minify.  So, the quickest build to use for rapid-testing during development very much uses this pattern.

Reply:
But why are you uploading your rapid test development symbols to Sentry? That's going to chew server disk space pretty quickly.

Reply:
Because the Sentry integration isn't done yet.  Need I remind you:

![Screen Shot 2020-05-28 at 12 21 58 PM](https://user-images.githubusercontent.com/25213402/83102074-e9443780-a0dd-11ea-98ba-cc2bb9d0a16c.png)

![Screen Shot 2020-05-28 at 12 22 21 PM](https://user-images.githubusercontent.com/25213402/83102080-ecd7be80-a0dd-11ea-9b1c-4f997c779b3c.png)

Granted, once it's done, I suppose it will generally be far less of an issue.  At least, unless we start worrying about custom scopes, deliberate breadcrumbing / event generation, etc.

Reply:
And, to call out myself to be fair:

![Screen Shot 2020-05-28 at 12 25 48 PM](https://user-images.githubusercontent.com/25213402/83102335-6a9bca00-a0de-11ea-9f4f-cb55562112d8.png)

Reply:
I'll clean it up a bit, but there will still be separate checks for embedded vs native.

Reply:
> Granted, once it's done, I suppose it will generally be far less of an issue.

Precisely. I don't want to have this kind of thing in build scripts. Design the script for the long term case. If uploading manually is a pain, just have a local script to do the sentry upload for your local testing.